src/Courses/Settings/OverridesForm.tsx

```diff
@@ -0,0 +1,238 @@
+import { Grid, TextField, Button, CircularProgress } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
+import { MaterialUiPickersDate } from '@material-ui/pickers/typings/date';
+import MomentUtils from '@date-io/moment';
+import moment, { Moment } from 'moment';
+import { extendQuestion, extendTopic, getQuestion, getQuestions, getTopic } from '../../APIInterfaces/BackendAPI/Requests/CourseRequests';
+import _ from 'lodash';
+import { NewCourseTopicObj, ProblemObject } from '../CourseInterfaces';
+
+type Inputs = {
+    startDate: Moment;
+    endDate: Moment;
+    deadDate: Moment;
+};
+
+interface OverridesFormProps {
+    userId: number;
+    topic?: NewCourseTopicObj;
+    problem?: ProblemObject;
+}
+
+export const OverridesForm: React.FC<OverridesFormProps> = ({topic, userId, problem}) => {
+    const { register, handleSubmit, watch, errors, control, setValue } = useForm<Inputs>();
+    const [formLoading, setFormLoading] = useState<boolean>(false);
+    const [submitLoading, setSubmitLoading] = useState<boolean>(false);
+    const drawerFontSize = '1.4em';
+
+    // Get Topic Override information
+    useEffect(()=>{
+        if (!topic || problem !== undefined) return;
+        setFormLoading(true);
+
+        (async () => {
+            try {
+                const res = await getTopic({id: topic.id, userId});
+                const topicData = res.data.data;
+                // If there are overrides for the selected user, overwrite the default dates in the object.
+                // Right now, we expect only one override per topic to be returned.
+                if (topicData.studentTopicOverride?.length === 1) {
+                    _.assign(topicData, topicData.studentTopicOverride[0]);
+                }
+
+                setValue('startDate', moment(topicData.startDate));
+                setValue('endDate', moment(topicData.endDate));
+                setValue('deadDate', moment(topicData.deadDate));
+            } catch (e) {
+                console.error(`Topic ${topic.id} or User ${userId} does not exist!`, e);
+            } finally {
+                setFormLoading(false);
+            }
+        })();
+    }, [topic, userId]);
```

Comment:
I'm not sure the solution here but watching an entire object isn't great
If there is a local change it causes an api call to fire off, you really only care if the id changes
I understand why you did this, it's because the linter yells at you, we just need to figure it out at some point

Reply:
A local change shouldn't matter unless it overwrites the object and it gets a new memory address.

Reply:
True, but even if that's the case if the id doesn't change I don't want to refetch